src/index.ts

```diff
@@ -42,7 +42,7 @@ export * from './config/to-type-sequence'
 export * from './config/condition'
 
 // Rules
-export { rule } from './config/rule'
+export { rule, RuleBuilder } from './config/rule'
```

Comment:
```suggestion
export type { BuildContext } from './utils/build-context'
export type { RuleBuilder } from './config/rule'
export { rule } from './config/rule'
```

Is `export type` enough? `BuildContext` may need be exported too.

Reply:
I think just RuleBuilder will do it... I'm using it with esm.sh so I'll confirm after the next release. Deno didn't like the require in `writeToProfile` so I'm using the below custom version. TypeScript is only complaining about `RuleBuilder` and I'm typing with a Karabiner config that `deno run -A index.ts` spit out now with no problems.

```ts
import {
  complexModifications,
  KarabinerConfig,
  ModificationParameters,
  Rule,
  RuleBuilder,
} from "https://esm.sh/karabiner.ts";

import { join } from "https://deno.land/std@0.192.0/path/mod.ts";

export async function writeToProfile(
  name: "--dry-run" | string,
  rules: Array<Rule | RuleBuilder>,
  parameters: ModificationParameters = {},
) {
  const karabinerConfigFile = join(
    Deno.env.get("HOME")!,
    ".config",
    "karabiner",
    "karabiner.json",
  );

  const config: KarabinerConfig = name === "--dry-run"
    ? { profiles: [{ name, complex_modifications: { rules: [] } }] }
    : (await import(karabinerConfigFile, { assert: { type: "json" } })).default;

  const profile = config?.profiles.find((v) => v.name === name);

  try {
    profile!.complex_modifications = complexModifications(rules, parameters);
  } catch (e) {
    console.error(e);
    return;
  }

  const json = JSON.stringify(config, null, 2);

  if (name === "--dry-run") {
    console.info(json);
    return;
  }

  Deno.writeTextFile(karabinerConfigFile, json);

  console.log(`✓ Profile ${name} updated.`);
}
```